x-pack/metricbeat/module/aws/cloudwatch/metadata/ec2/ec2.go

```diff
@@ -55,7 +55,7 @@ func AddMetadata(regionName string, awsConfig awssdk.Config, fips_enabled bool,
 			if output.InstanceType != "" {
 				_, _ = events[eventIdentifier].RootFields.Put("cloud.machine.type", output.InstanceType)
 			} else {
-				logp.Error(fmt.Errorf("InstanceType is empty"))
+				logp.L().Error("InstanceType is empty")
```

Comment:
This does not reuse an existing logger named "cloudwatch", but I think it is better to indentify where the error happened for the end-user.

I dived a bit into `zap` (on which `logp` is based) and from my understanding:

`L()` calls the `loadLogger` function which returns the `coreLogger` global instance (hence no copy)

- https://github.com/elastic/elastic-agent-libs/blob/ee5dad913b28384320be73519fb255457046b8b3/logp/logger.go#L233
- https://github.com/elastic/elastic-agent-libs/blob/ee5dad913b28384320be73519fb255457046b8b3/logp/core.go#L269

`Named(string)` makes a copy of the logger and adds a name to it.

- https://github.com/uber-go/zap/blob/845ca51d5b8d9fed9fe14f35ab13b6b160d5762d/logger.go#L150
- https://github.com/uber-go/zap/blob/master/logger.go#L284

Reply:
Okay, I got it. 

So this `Named()` function is meant to be used as a factory to create a reusable logger, and it may allocate a lot of structures if the error is repeated.